src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/RObjectEntry.java

```diff
@@ -96,9 +98,29 @@ public String getDisplayValue()
                       rObject.getDescription().trim() :
                       val;
    }
+   
+   private boolean rObjectHasDataClass()
+   {
+      JsArrayString classes = rObject.getClazz();
+      for (int i = 0, n = classes.length(); i < n; i++)
+         if (DATA_CLASSES.contains(classes.get(i)))
+            return true;
+      return false;
+   }
 
    public static final String NO_VALUE = "NO_VALUE";
-
+   
+   private static final Set<String> DATA_CLASSES = new HashSet<String>();
+   static {
+      DATA_CLASSES.addAll(ListUtil.create(
```

Comment:
Should this include and/or share the elements of this list? https://github.com/rstudio/rstudio/blob/54cd3abcfc58837b433464c793fe9b03a87f0bb4/src/gwt/src/org/rstudio/studio/client/workbench/views/environment/view/RObjectEntry.java#L69-L74